version: "3.8"
services:
  backend-flask:
    hostname: backend-flask
    container_name: backend-flask
    environment:
      FRONTEND_URL: "http://frontend-react-js:3000"
      BACKEND_URL: "http://backend-flask:4567"
    build: ./backend-flask
    ports:
      - "4567:4567"
    volumes:
      - ./backend-flask:/backend-flask
  frontend-react-js:
    hostname: frontend-react-js
    container_name: frontend-react-js
    environment:
      REACT_APP_BACKEND_URL: "http://backend-flask:4567"
    build: ./frontend-react-js
    ports:
      - "3000:3000"
    #volumes:
    #  - ./frontend-react-js:/frontend-react-js
  db:
    hostname: db
    container_name: db
    image: postgres:13-alpine
    restart: always
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password
    ports:
      - '5432:5432'
    volumes: 
      - db:/var/lib/postgresql/data
  dynamodb-local:
    hostname: dynamodb-local
    container_name: dynamodb-local
    # https://stackoverflow.com/questions/67533058/persist-local-dynamodb-data-in-volumes-lack-permission-unable-to-open-databa
    # We needed to add user:root to get this working.
    user: root
    command: "-jar DynamoDBLocal.jar -sharedDb -dbPath ./data"
    image: "amazon/dynamodb-local:latest"
    ports:
      - "8000:8000"
    volumes:
      - "./docker/dynamodb:/home/dynamodblocal/data"
    working_dir: /home/dynamodblocal

# the name flag is a hack to change the default prepend folder
# name when outputting the image names
networks: 
  internal-network:
    driver: bridge
    name: cruddur

volumes:
  db:
    driver: local
